lib/pluginmanager/bundler/logstash_injector.rb

```diff
@@ -33,16 +34,23 @@ def inject(gemfile_path, lockfile_path)
       end
 
       builder = Dsl.new
-      builder.eval_gemfile(gemfile_path)
-
-      @new_deps -= builder.dependencies
-
-      builder.eval_gemfile("injected gems", new_gem_lines) if @new_deps.any?
-      definition = builder.to_definition(lockfile_path, {})
-      append_to(gemfile_path) if @new_deps.any?
-      definition.lock(lockfile_path)
-
-      return @new_deps
+      gemfile = LogStash::Gemfile.new(File.new(gemfile_path, "r+")).load
```

Comment:
This is taken care by `LogStash::Gemfile#save` and `LogStash::Gemfile#restore!` that we call in the rescue block.